src/components/ListItemCard.tsx

```diff
@@ -0,0 +1,77 @@
+/** @format */
+
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import Typography from "@mui/material/Typography";
+import { useNavigate } from "react-router-dom";
+import { useContext } from "react";
+
+import {
+	Button,
+	CardActionArea,
+	CardActions,
+} from "@mui/material";
+import "../assets/data.json";
+import { PokemonContext } from "../context/PokemonContext";
+
+type Pokemon = {
+	name: string;
+	url: string;
+};
+
+const ListItemCard = (
+	props: Pokemon
+) => {
+	const navigate = useNavigate();
+	const pokemonContext = useContext(
+		PokemonContext
+	);
+	const navigateToDetail = (
+		selectedName: string,
+		selectedUrl: string
+	) => {
+		pokemonContext?.loadDetail(
+			selectedUrl
+		);
+		pokemonContext?.setSelectedPokemon(
+			selectedName
+		);
```

Comment:
This wouldn't have been necessary if dynamic routing were used

Reply:
fixed